diff -Naurx data a/plugin.py b/plugin.py
--- a/plugin.py	2015-05-15 17:20:17.000000000 +0200
+++ b/plugin.py	2015-05-15 22:07:06.283887027 +0200
@@ -206,7 +206,7 @@
         self.close()
 
     def _cbCheckXBMCTimer(self):
-        data = os.popen('ps -ef | grep xbmc.bin | grep -v grep').read().strip()
+        data = os.popen('ps | grep xbmc.bin | grep -v grep').read().strip()
 	if data is None or data == "":
 	    self.checkXBMCTimer.stop()
             os.system('xbmc.helper --stop')
