From 7824c5ec8c43daf1972d8ff9ee553c56cc06f0d4 Mon Sep 17 00:00:00 2001
From: meo <lupomeo@hotmail.com>
Date: Fri, 5 Jun 2015 22:47:48 +0200
Subject: [PATCH] Black Hole hacks

---
 src/OMBManagerInstall.py |   19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/src/OMBManagerInstall.py b/src/OMBManagerInstall.py
index b13b237..6556a0b 100644
--- a/src/OMBManagerInstall.py
+++ b/src/OMBManagerInstall.py
@@ -343,7 +343,24 @@ class OMBManagerInstall(Screen):
 		os.system(OMB_UMOUNT_BIN + ' ' + ubi_path)
 		os.system(OMB_UBIDETACH_BIN + ' -m ' + mtd)
 		os.system(OMB_RMMOD_BIN + ' nandsim')
-
+		
+		if not os.path.exists('/usr/lib/python2.7/boxbranding.so'):
+			os.system("ln -s /usr/lib/enigma2/python/boxbranding.so /usr/lib/python2.7/boxbranding.so")
+		if os.path.exists(dst_path + '/usr/lib/python2.7/boxbranding.py'):
+			os.system("cp /usr/lib/enigma2/python/boxbranding.so " + dst_path + "/usr/lib/python2.7/boxbranding.so")
+			os.system("rm -f " + dst_path + '/usr/lib/python2.7/boxbranding.py')
+		if dst_path.find('OpenPLi') != -1:
+			inme = open(dst_path + "/etc/init.d/volatile-media.sh",'r')
+			out = open(dst_path + "/etc/init.d/volatile-media.tmp",'w')
+			for line in inme.readlines():
+				if line.find('mount -t tmpfs -o size=64k tmpfs /media') != -1:
+					line = "mountpoint -q \"/media\" || mount -t tmpfs -o size=64k tmpfs /media\n"
+				out.write(line)
+			out.close()
+			inme.close()
+			os.system("mv -f " + dst_path + "/etc/init.d/volatile-media.tmp " + dst_path + "/etc/init.d/volatile-media.sh"  )
+			os.system("chmod 0755 " + dst_path + "/etc/init.d/volatile-media.sh")
+		
 		return True
 
 	# Based on nfi Extract by gutemine
-- 
1.7.9.5

